Test/Core/Cloning/SpecificCloningTest.cs

```diff
@@ -40,25 +43,53 @@ public class SpecificCloningTest
 			Assert.AreEqual(source.Options, target.Options);
 			Assert.AreEqual(source.MatchTimeout, target.MatchTimeout);
 		}
-		[Test] public void CloneCultureInfo()
+		[Test]
+		public void CloneCultureInfo()
 		{
 			CultureInfo source = new CultureInfo("en-US");
 			CultureInfo target = source.DeepClone();
 
 			Assert.AreNotSame(source, target);
 			Assert.AreEqual(source, target);
 		}
-		[Test] public void CloneHashSet()
+		[Test]
+		public void CloneHashSet_ValueType()
 		{
 			Random rnd = new Random();
-			HashSet<int> source = new HashSet<int>(Enumerable.Range(0, 50).Select(i => rnd.Next()));
+			var values = Enumerable.Range(0, 50).Select(i => rnd.Next());
+			HashSet<int> source = new HashSet<int>(values);
 			HashSet<int> target = source.DeepClone();
 
 			Assert.AreNotSame(source, target);
-			CollectionAssert.AreEquivalent(source, target);
+			CollectionAssert.AreEqual(source, target);
```

Comment:
`AreEquivalent` should be enough here. A hashset doesn't make any guarantees about element order and cloning a hashset would be allowed to change order, as long as the same elements are present.